sdk/services/prices.ts

```diff
@@ -219,6 +201,70 @@ export default class PricesService {
 		return offChainPrices;
 	}
 
+	private connectToPyth(networkId: NetworkId, server: PriceServer) {
+		if (this.pyth) {
+			this.pyth.closeWebSocket();
+		}
+
+		this.pyth = new EvmPriceServiceConnection(getPythNetworkUrl(networkId, server), {
+			logger: LOG_WS ? console : undefined,
+		});
+
+		this.pyth
+			.getPriceFeedIds()
+			.then((_) => {
+				this.sdk.context.events.emit('prices_connection_update', {
+					connected: true,
+				});
+			})
+			.catch((error) => {
+				this.sdk.context.events.emit('prices_connection_update', {
+					connected: false,
+					error: error || new Error('pyth prices ws connection failed'),
+				});
+			});
+	}
+
+	private setEventListeners() {
+		this.sdk.context.events.on('network_changed', (params) => {
+			if (this.pyth) {
+				this.pyth.closeWebSocket();
+			}
+			this.connectToPyth(params.networkId, this.server);
+		});
+
+		this.sdk.context.events.on('prices_connection_update', (params) => {
+			if (params.connected) {
+				this.subscribeToPythPriceUpdates();
+			} else {
+				this.retryConnection(this.sdk.context.networkId);
+			}
+		});
+
+		if (this.pyth) {
+			this.pyth.onWsError = (error: Error) => {
```

Comment:
I think we need to make this call where the pyth instance is initialized otherwise if we create a new instance this setEventListeners would not have been called again and we won't get this onError callback.

Reply:
Yes. That's true. Moved this to the `connectToPyth`.